### PR TITLE
Improved errorlog  in add-label.js #2438

### LINKED ISSUE #2438: GitHub Actions: Improvements to error logs in add-label.js
### Overview
As a developer, we need to ensure that the GHA to add labels produces a more descriptive error log when failing because currently it produces a generic error statement which might not be too helpful while debugging.

### Action Items
- [x] Become familiar with how the [GitHub action](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/add-update-label-weekly.yml) "Add Update Label to Issues Weekly" works
- [x] [JS Edits](https://github.com/hackforla/website/blob/gh-pages/github-actions/add-update-label-weekly/add-label.js)
    - [x] Under the function removeLabels (line 141), update the catch so it logs the exact error when it fails
    - [x] Under the function addLabels (line 163), update the catch so it takes in err as a parameter, and log the exact error if it fails
    - [x] Under the function postComment (line 190), update the catch so it logs the exact error when it fails
    - [x] Under the function getAssignees (line 227), update the catch so it logs the exact error when it fails

### Details
It is always better to have more info on what the specific error is, rather than outputting a generic statement that a developer won't be able to debug. At the least, the error code should be added. At most, just include the body of the error statement so a developer can have an idea of why something failed.
<details>
<summary>Click for an example for a good error logging</summary>

```
console.err(`Function failed to <insert specific action here>. Please refer to the error below: \n ${err}`)
```

</details>

### Resources/Instructions
[add-update-label-weekly](https://github.com/hackforla/website/tree/gh-pages/github-actions/add-update-label-weekly)

### PR DESCRIPTION
**Pull request by Suman2795**
Fixes #2438 

### What changes did you make and why did you make them ?
   Made improvements to the error logs in add-label.js file

  - In line no 214, for function removeLabels updated the catch statement to retrieve the error
  -In line no 235, for function addLabels updated the catch statement to retrieve the error
  -In line no 249, for function postComment updated the catch statement to retrieve the error
  -In line no 283, for function getAssignees updated the catch statement to retrieve the error

### CODE REVIEW

**Comment by t-will-gillis** [comment]
Review ETA: 5 PM 10/25
Availability: 10-5 Tuesday

**Comment by jyaymie** [comment]
Review ETA: 1pm 10/24/22
Availability: Now through 1pm

File: github-actions/add-update-label-weekly/add-label.js [hunk 1]
```diff
@@ -232,7 +232,7 @@ async function addLabels(issueNum, ...labels) {
     console.log(`Added these labels to issue #${issueNum}: ${labels}`);
     // If an error is found, the rest of the script does not stop.
   } catch {
```
Comment (jyaymie) [inline github-actions/add-update-label-weekly/add-label.js]: I believe the catch needs to take in `err` as a parameter.

File: github-actions/add-update-label-weekly/add-label.js [hunk 2]
```diff
@@ -211,7 +211,7 @@ async function removeLabels(issueNum, ...labels) {
       });
       console.log(`Removed "${label}" from issue #${issueNum}`);
     } catch (err) { 
-      console.error(`No "${label}" label to remove for issue #${issueNum}`);
+      console.error(`Function failed to remove labels.Please refer to the error below: \n`,err);
```
Comment (jyaymie) [inline github-actions/add-update-label-weekly/add-label.js]: It's a good idea to add a space between the two sentences to make it more readable. I would also add a space between the comma and `err`. (These recommendations apply to all the error messages that were changed.)

File: github-actions/add-update-label-weekly/add-label.js [hunk 3]
```diff
@@ -232,7 +232,7 @@ async function addLabels(issueNum, ...labels) {
     console.log(`Added these labels to issue #${issueNum}: ${labels}`);
     // If an error is found, the rest of the script does not stop.
   } catch {
-    console.error(`Could not add these labels for issue #${issueNum}: ${labels}`);
+    console.error(`Function failed to add labels.Please refer to the error below:\n`,err);
```
Comment (jyaymie) [inline github-actions/add-update-label-weekly/add-label.js]: This is probably me just being nit-picky, so this one's up to you! I notice an inconsistency with the error messages, particularly with the spacing around `\n`. I recommend sticking with one format and using it for all your error messages.

**Review by jyaymie — CHANGES_REQUESTED** [review]
Fantastic work, @Suman2795! You linked the correct issue and found the appropriate functions to modify despite the difference in line numbers. There are just a few things I noticed, which I marked with inline comments. Let me know if you think otherwise!

File: github-actions/add-update-label-weekly/add-label.js (hunk 2 above)
Comment (t-will-gillis) [reply github-actions/add-update-label-weekly/add-label.js]: I would also opt for the space between the period at the end of the first sentence and the start of the second. (As far as the spaces at  "    below: \n', err  " , my only observation is that I would opt for consistency- whichever way you choose to use spaces in all four of the changes) 


_<<I previously included a comment regarding the formatting of the console.error() message but have removed it- Will>>_

File: github-actions/add-update-label-weekly/add-label.js (hunk 1 above)
Comment (t-will-gillis) [reply github-actions/add-update-label-weekly/add-label.js]: I also believe the `err` parameter is needed

**Review by t-will-gillis — CHANGES_REQUESTED** [review]
Hi @Suman2795 - Suman great work with the error log edits and tracking down the correct line numbers! I am adding a couple of comments to Jamie's, which I hope are useful. Thanks!

File: github-actions/add-update-label-weekly/add-label.js (hunk 1 above)
Comment (t-will-gillis) [reply github-actions/add-update-label-weekly/add-label.js]: I see the (err) parameter at Line 234 in the updated code- looks good thanks!

File: github-actions/add-update-label-weekly/add-label.js (hunk 2 above)
Comment (t-will-gillis) [reply github-actions/add-update-label-weekly/add-label.js]: Hi @Suman2795 - Thank you for the adding the space between sentences and making the four console.error() changes consistent in construction- very 'Clean'. This looks great.

**Review by t-will-gillis — APPROVED** [review]

**Comment by MattPereira** [comment]
Review ETA: End of Day 10/27/22
Availability: 2 hours

**Review by MattPereira — APPROVED** [review]
Excellent work @Suman2795 ! :+1: 

You updated the catch blocks to log the exact error when the parent function fails, which will be much more helpful for debugging. Additionally, I see you made all of the changes requested by @jyaymie and @t-will-gillis and now all of the `console.error`'s have matching spacing. 

Good teamwork everyone!

**Review by jyaymie — APPROVED** [review]

**Comment by esantiano** [comment]
Review ETA: 10/29/22
Availability: 3 - 5PM

**Review by esantiano — APPROVED** [review]
Hey @Suman2795,

The changes you made look nice and consistent, its a lot easier to understand what specific functions are failing and why. Nice job!

**Pull request merged**